src/aviary/env.py

```diff
@@ -234,11 +237,30 @@ async def _exec_tool_call(tool_call: ToolCall) -> ToolResponseMessage:
                 s_content = json.dumps(content)
             return ToolResponseMessage.from_call(tool_call, content=s_content)
 
+        invalid_msgs = []
+        valid_action = message
+        if handle_invalid_tool_calls:
+            valid_action, invalid_action = self.filter_invalid_tool_calls(message)
```

Comment:
We need to preserve the ordering of tool calls here I think, so we'll need to prolly make some changes to `filter_invalid_tool_calls`

Reply:
FWIW OpenAI and Anthropic don't enforce ordering (though I am not sure if there's a performance hit). It can matter for some local models though. I'm ambivalent, but we could circle back to this if it'd require a refactor to support ordering.

Reply:
fixed